src/Microsoft.AspNetCore.SystemWebAdapters/HttpCookie.cs

```diff
@@ -68,37 +76,38 @@ public NameValueCollection Values
     {
         get
         {
-            if (_holder is HttpValueCollection values)
+            if (_multiValue == null)
             {
-                return values;
+                // create collection on demand
+                _multiValue = new HttpValueCollection();
+
+                // convert existing string value into multivalue
+                if (_stringValue != null)
+                {
+                    if (_stringValue.Contains('&', StringComparison.InvariantCulture) || _stringValue.Contains('=', StringComparison.InvariantCulture))
```

Comment:
is this what system.web was doing?

Reply:
@twsouthwick no, System.web is doing an `.IndexOf` without the `SystemComparison` argument. Do you want me to change it to IndexOf?

Reply:
No this is fine -I was wondering if the rest of it was something that system.web was doing